learntools/machine_learning/nbs/ch6.ipynb

```diff
@@ -0,0 +1,172 @@
+    "import pandas as pd\n",
+    "from sklearn.metrics import mean_absolute_error\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "from sklearn.tree import DecisionTreeRegressor\n",
+    "\n",
+    "\n",
+    "# Path of the file to read\n",
+    "iowa_file_path = '../input/house-prices-advanced-regression-techniques/train.csv'\n",
+    "\n",
+    "home_data = pd.read_csv(iowa_file_path)\n",
+    "# Create target object and call it y\n",
+    "y = home_data.SalePrice\n",
+    "# Create X\n",
+    "predictor_columns = ['LotArea', 'YearBuilt', '1stFlrSF', '2ndFlrSF', 'FullBath', 'BedroomAbvGr', 'TotRmsAbvGrd']\n",
+    "X = home_data[predictor_columns]\n",
+    "\n",
+    "# Split into validation and training data\n",
+    "train_X, val_X, train_y, val_y = train_test_split(X, y, random_state=1)\n",
+    "\n",
+    "# Specify Model\n",
+    "iowa_model = DecisionTreeRegressor(random_state=1)\n",
+    "# Fit Model\n",
+    "iowa_model.fit(train_X, train_y)\n",
+    "\n",
+    "# Make validation predictions and calculate mean absolute error\n",
+    "val_predictions = iowa_model.predict(val_X)\n",
+    "val_mae = mean_absolute_error(val_predictions, val_y)\n",
+    "print(\"Validation MAE when not specifying max_leaf_nodes: {:,.0f}\".format(val_mae))\n",
+    "\n",
+    "# Using best value for max_leaf_nodes\n",
+    "iowa_model = DecisionTreeRegressor(max_leaf_nodes=100, random_state=1)\n",
+    "iowa_model.fit(train_X, train_y)\n",
+    "val_predictions = iowa_model.predict(val_X)\n",
+    "val_mae = mean_absolute_error(val_predictions, val_y)\n",
+    "print(\"Validation MAE for best value of max_leaf_nodes: {:,.0f}\".format(val_mae))\n",
+    "\n",
+    "\n",
+    "# Set up code checking\n",
+    "from learntools.core import binder\n",
+    "binder.bind(globals())\n",
+    "from learntools.machine_learning.ex6 import *\n",
+    "print(\"\\nSetup complete\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Exercises\n",
+    "Data science isn't always this easy. But replacing the decision tree with a Random Forest is going to be an easy win."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 1: Use a Random Forest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Validation MAE for Random Forest Model: \n"
+     ]
+    }
+   ],
+   "source": [
+    "from sklearn.ensemble import RandomForestRegressor\n",
+    "\n",
+    "# Define the model. Set random_state to 1\n",
+    "rf_model = _\n",
+    "\n",
+    "# fit your model\n",
+    "_\n",
+    "\n",
+    "# Calculate the mean absolute error of your Random Forest model on the validation data\n",
+    "rf_val_mae = _\n",
+    "\n",
+    "print(\"Validation MAE for Random Forest Model: {}\".format(rf_val_mae))\n",
+    "\n",
+    "# step_1.check()"
```

Comment:
Also, I've been following the convention of always starting with the call to `check()` uncommented.